middleware/rpc/provider.go

```diff
@@ -288,6 +330,17 @@ func Provider(server rpc.ResourceProviderServer) p.Provider {
 				return p.CallResponse{}, err
 			}
 
+			// upgrade the return dependencies if the provider doesn't support output values,
+			// in which case [rpcResp.ReturnDependencies] has meaningful information.
+			_return := resp.Return.AsMap()
+			for name, deps := range rpcResp.GetReturnDependencies() {
+				if v, ok := _return[name]; ok {
+					deps := mergePropertyDependencies(append(v.Dependencies(), putil.ToUrns(deps.GetUrns())...))
```

Comment:
Good catch! Property dependencies should just do this: https://github.com/pulumi/pulumi/pull/19423.